guardian/prosemirror-invisibles
# Toggle decorations

NB: Depends on #24 – please review this first!

## What does this change?

Adds state to toggle plugin decorations on and off, a likely feature in most RTEs. IIUC the alternative at the moment is to unmount the plugin completely.

![invisibles-toggle](https://user-images.githubusercontent.com/7767575/103319042-49aaa700-4a28-11eb-9263-b4a01c13516c.gif)

## How to test

The automated tests should pass.

Launch the `pages` demo. I've added a checkbox to toggle the invisible chars on and off – it should work as above.

## Dev notes

There's more code than you might expect here! Now that we're storing more than the plugin decorations themselves, I've added a reducer to handle the plugin state. This might be overkill at this point, but this pattern has been effective [elsewhere](https://github.com/guardian/prosemirror-typerighter)), and does make it easier to add and test additional state in future if required.
